food-truck-frontend/src/components/Form.tsx

```diff
@@ -103,6 +103,17 @@ class Form extends Component<Props, State> {
             }
         }));
     }
+
+    static getDerivedStateFromProps(props: Props, state: State) {
+        let newFormData: any = {};
+        if (props.children)
+            React.Children.forEach(props.children, (c: any) => {
+                let name: string = (c as ReactElement).props.name as string;
+                newFormData[name] = state.formData[name];
+            });
+        state.formData = newFormData;
+        return state;
+    }
```

Comment:
What is this for?

Reply:
This is to solve an issue with the Form not wanting to dynamically alter its content. The way React optimizes, it doesn't re-instantiate components when props change, it just re-runs the lifecycle methods. Since the state was formalized in the constructor, this meant that, should a form entry component be removed (for me, toggling extended to non-extended review dropping the text field for entering the review), its data would still exist in the form and be submitted (thus making it impossible to submit a non-extended review after saving an extended one).

As for what the method does, this is a life cycle method that runs when the props are updated, allowing you to update state based on it. While there are MANY anti-patterns revolving around this method, our use case (filtering out any state attributes that no longer existed in the form) fell under one of the few legitimate uses.

I'll confess there may still be a bug or two with it (I'm pretty sure adding a new component worked fine, since I could submit extended reviews in the first place when it defaulted to non-extended, but I can't swear to it). But I probably didn't test it enough.